### ISSUE TITLE
Sublime Text 3 bracketHighlighter dont know how to color bracket underlines

### ISSUE THREAD

**Issue by Mytex** [facelessuser/BracketHighlighter#363]
I dont know how to change the underline color of the brackets using brackethiglighter i did check http://facelessuser.github.io/BracketHighlighter/customize/#configuring-highlight-style but i dont understand it im new to Sublime Text 3 and BH

**Comment by r-stein** [comment]
1. Install [PackageResourceViewer](https://packagecontrol.io/packages/PackageResourceViewer)
2. Press `ctrl+shift+p` and write `PackageResourceViewer: Open Resource`
3. Select your colorscheme (it should be under `Color Scheme - Default` if you use a default scheme)
4. Add this at a reasonable position and save (You can change the foreground color):
``` xml
        <dict>
            <key>name</key>
            <string>Bracket Default</string>
            <key>scope</key>
            <string>brackethighlighter.default</string>
            <key>settings</key>
            <dict>
                <key>foreground</key>
                <string>#CC99CC</string>
            </dict>
        </dict>
```
5.. If you use SublimeLinter or other packages you might need to reselect the colorscheme

**Comment by facelessuser** [comment]
You can see @r-stein is defining a scope called `brackethighlighter.default` in his color scheme and assigning the color `#CC99CC` to that scope.  You can also see that the default bracket style assigns that scope to the `color` key.

You can also change the default style or another predefined style to match a scope already in your theme if you don't want to modify your theme.   Just take a look at the scope colorization rules in your color scheme and pick a scope that works for you. The key name `color` is the scope.  `color` is the name because the scope determines the color.  ColorSchemes colorize certain scopes, so to colorize the regions with underlines or whatever, we assign a scope, and the color scheme colorizes that scope.   That is just how Sublime requires regions to be defined. 

```js
        // "default" style defines attributes that
        // will be used for any style that does not
        // explicitly define that attribute.  So if
        // a style does not define a color, it will
        // use the color from the "default" style.
        "default": {
            "icon": "dot",
            "color": "brackethighlighter.default",
            "style": "underline"
        },

        // This particular style is used to highlight
        // unmatched bracket pairs.  It is a special
        // style.
        "unmatched": {
            "icon": "question",
            // "color": "brackethighlighter.unmatched",
            "style": "outline"
        },
        // User defined region styles
        "curly": {
            "icon": "curly_bracket"
            // "color": "brackethighlighter.curly",
            // "style": "underline"
        },
        "tag": {
            "icon": "tag",
            // "endpoints": true,
            // "color": "brackethighlighter.tag",
            "style": "outline"
        },
```

If you have more questions, please be *specific* about what you don't understand in our documentation as we are always looking to improve the documentation for people.

@r-stein  Thanks for for your help on the issue.  It is always appreciated :smile:.

**Comment by Mytex** [comment]
facelessuser i found bh_core.sublime-settings and i found 
"default": {
            "icon": "dot",
            "color": "brackethighlighter.default",
            "style": "underline"
        },

        // This particular style is used to highlight
        // unmatched bracket pairs.  It is a special
        // style.
        "unmatched": {
            "icon": "question",
            // "color": "brackethighlighter.unmatched",
            "style": "outline"
        },
        // User defined region styles
        "curly": {
            "icon": "curly_bracket"
            // "color": "brackethighlighter.curly",
            // "style": "underline"
        },
        "tag": {
            "icon": "tag",
            // "endpoints": true,
            // "color": "brackethighlighter.tag",
            "style": "outline"
        
Now how do i make it so the underline of the brackets is blue

**Comment by facelessuser** [comment]
You use the scope that is highlighted blue in your color scheme.  If nothing is highlighted blue in your color scheme, you have to add a new color rule in your color scheme and define the scope that you will use.

For instance, since I don't know what color scheme you use and what scope in that theme gets highlighted a shade of blue, if we add a rule like this to our color scheme, and we use that defined scope for bracket style, we should bet blue as `#0000FF` is blue.

```xml
<dict>
            <key>name</key>
            <string>Bracket Default</string>
            <key>scope</key>
            <string>brackethighlighter.default</string>
            <key>settings</key>
            <dict>
                <key>foreground</key>
                <string>#0000FF</string>
            </dict>
        </dict>
```

**Comment by Mytex** [comment]
its very difficult for me i cant even understand some stuff u guys have typed ;-;

**Comment by Mytex** [comment]
i still cant do it and please facelessuser respond or reply faster i cant understand you are typing about stuff dat i dont know

**Comment by r-stein** [comment]
Have you tried to follow my stepwise manual? It describes how to edit your colorscheme. If you did not change it, then it will be under `Color Scheme - Default/Monokai.tmTheme`.

**Comment by Mytex** [comment]
<dict>
            <key>name</key>
            <string>Bracket Default</string>
            <key>scope</key>
            <string>brackethighlighter.default</string>
            <key>settings</key>
            <dict>
                <key>foreground</key>
                <string>#CC99CC</string>
            </dict>
        </dict>


Doesnt do anything

**Comment by facelessuser** [comment]
I think you should spend some time reading about color schemes: http://docs.sublimetext.info/en/latest/reference/color_schemes.html.

> please facelessuser respond or reply faster

I do this in my free time, so I respond when I have time and when I am not doing other things.  I do not drop everything to offer support.

**Comment by facelessuser** [comment]
> Doesnt do anything

It is possible you might not have put it in the correct space.

- You will have to provide us with the full color scheme content so we can verify you have done it correctly.
- You will also need to provide us us with any modifications you have made to your settings file (if any).
- You will also have to identify for us which bracket you are using as your reference to determine that it is working.

**Comment by Mytex** [comment]
http://i.imgur.com/5cTuwny.png      http://i.imgur.com/HzfYSuQ.png

**Comment by r-stein** [comment]
Your bhcore settings file must be encapsulated by `{..}`, i.e. add a `{` a the start and a `}` at the end.

In the colorscheme tmTheme file you need to move it inside the plist attribute. Move it between an closing `</dict>` and an opening `<dict>` tag of the same indentation.

**Comment by Mytex** [comment]
about http://i.imgur.com/8wl5Fgk.png i dont understand i dont know what to do and about the one http://i.imgur.com/7meU6JS.png

**Comment by r-stein** [comment]
In http://i.imgur.com/7meU6JS.png you add a line, with the content `{` above line 1 and a line, with the content `}` below the last line.

and change http://i.imgur.com/8wl5Fgk.png to this:
![bh_example](https://cloud.githubusercontent.com/assets/12573621/17749972/4cc0059a-64c0-11e6-819f-cca2bed1584e.png)

**Comment by Mytex** [comment]
well this happened http://i.imgur.com/QsFWuEK.png and also http://i.imgur.com/9I8gXJ2.png

**Comment by r-stein** [comment]
You are missing a `>` behind `</dict` in line 567

**Comment by Mytex** [comment]
I did this http://i.imgur.com/j7829iT.png and this http://i.imgur.com/ZeYcSq9.png but i the underlines of the brackets havent changed http://i.imgur.com/j9EPfDW.png

**Comment by r-stein** [comment]
Bracket Highlighter is disabled by default in Plain Text files.
Add this to your `bh_core.sublime-settings` file in line 2 under the opening `{`:
``` js
    "user_brackets": [
        {
            "name": "curly",
            "language_list": ["Hex"],
        },
        {
            "name": "round",
            "language_list": ["Hex", "RTF"],
        },
        {
            "name": "square",
            "language_list": ["Hex", "RTF"],
        },
    ],
```

**Comment by Mytex** [comment]
Thank you soo much you 2 helped me alot im very thankful to both of you

**Comment by Mytex** [comment]
erm how do i change the bracket underline size or thickness

**Comment by facelessuser** [comment]
You don't really have control over the bracket size and thickness per se.   What you can do is change the style to one of Sublime's allowed preset styles.  By default we use `underline` and I think `outline` for tags.  All available styles are listed here: http://facelessuser.github.io/BracketHighlighter/customize/#configuring-highlight-style.

**Comment by Mytex** [comment]
i didnt type bracket size and thickness i typed bracket underline size and thickness

**Comment by facelessuser** [comment]
Just replace bracket with bracket underline. That is what I meant.

**Comment by facelessuser** [comment]
So it looks like the original question was answered, so I will close this issue.

**Comment by MakhouT** [comment]
I'm adding an other question here as it's also regarding visuals. I didn't want to open a new issue for this. If you'd rather me open a new issue let me know. 

Basically I'm trying to figure out if what I am trying to do is possible with BH. Currently I am using the experimental feature content_highlight_bar. But would it be possible to have the content hightlight inside the brackets highlighter. 

Reason for this is:
- You'll have a bit of a thicker line
- less aggresive
- "not in the way"
- and because I'm used to it from an other IDE (jDeveloper) I use on a day to day basis 

I'll paste in a screenshot of the IDE I'm using with the highlighting in action, as that will make more sense than my explanation. 
![capture](https://cloud.githubusercontent.com/assets/3041165/17805501/1b45bafa-6600-11e6-98f9-be977302ea1c.PNG)

Basically it's that lightblue color, I would like to achieve that in ST with BH. But have the highligting between the brackets of BH on the left hand side from the line numbers.

**Comment by r-stein** [comment]
@MakhouT In contrast to Atom the Sublime Text API is very limited on customizing the GUI. (The API is small and fast, but has nearly everything one needs.). As far I know one could only add icons left of the numbers, where the other bracket symbols are already added.

In short: no that is not possible in Sublime Text.

**Comment by MakhouT** [comment]
@r-stein Ok thanks for the info. Was hoping it was just adding a background-color somewhere :)

**Comment by facelessuser** [comment]
> As far I know one could only add icons left of the numbers, where the other bracket symbols are already added.

That is correct, we can't really change the background of the gutter, but we can use icons.  I once experimented with having a visual indicator like the highlight line in the gutter, but unfortunately since other plugins also fight for the gutter, it turned out pretty bad.

@MakhouT Also, please do create separate issues for questions like this.  This is mainly so others can search for answers to the same questions, and so I can search and direct people to the same question.
